ios/FluentUI/Core/Theme/Tokens/TokenizedControlConfigurable.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Internal protocol for all controls' internal `configuration` objects.
+protocol ControlConfiguration: NSObject, ObservableObject, Identifiable {
+    associatedtype TokenType: ControlTokens
+
+    /// Custom design token set for this control, to use in place of the control's default Fluent tokens.
+    var overrideTokens: TokenType? { get set }
+
+    /// Common token lookup method, which should contain a lookup via the current environment's `fluentTheme`.
+    var tokens: TokenType { get set }
+}
+
+/// SwiftUI-specific extension to `TokenizedControl` that adds information about `configuration`.
+protocol TokenizedControlConfigurable: TokenizedControlInternal {
+    associatedtype ConfigurationType: ControlConfiguration where ConfigurationType.TokenType == TokenType
+
+    /// Contains additional configuration information about the control.
+    var state: ConfigurationType { get }
```

Comment:
should it still be "state"?

Reply:
Fair question. I could switch it to `configuration` now, but that's gonna be a big, boring change that touches dozens of files across everything. I'm worried that the noise of that change will drown out the signal of what I want to be reviewed here: the splitting-up of the `TokenizedControlInternal` protocol.

So ideally I'd prefer to have a separate PR that is just that single, mechanical change-over.